### ISSUE TITLE
Backwards forwards capability OWF/OZP environments

### ISSUE THREAD

**Issue by oneilos77** [ozone-development/ozp-community-request#13]
Navy requires OWF widgets moved into OZP environments to function as they did in OWF environment

**Comment by tonyalletag** [comment]
Navy requires IWC between OWF/OZP widgets

**Comment by tonyalletag** [comment]
Discussion of Options:
1) update the legacy widget adapter to include pub-sub eventing and continue to support and update the legacy widget adapter
* This require a developer to constantly update and maintain the adapter so that it can keep up with changes implemented in ozp-iwc.
2) Develop a plan to update legacy OWF widget to use the OZP-IWC libraries
* Only widgets that use OWF IWC would need to be updated
3) Build a custom webtop that allow OWF and OZP widgets to communicate

**Comment by itaran** [comment]
I don't believe that maintaining an adapter is the right way to go.  It will be struggle to keep the adapter aligned and once we have developers reliant on the adapter, we'll probably have to maintain it until the end of time.  This requires commitment and funding, something no one can reasonably commit to at this time.

Instead, why not look at building custom webtops.  Here are some options:
- Start with OWF7.X webtop as a base and modify it to instantiate the new OZP-IWC through an invisible iframe.
- Start with OZP webtop and modify it to include the old OWF pub/sub libraries
- Build a brand new webtop to have whatever features you want it to have and if possible make it FOSS for others to use.
